src/services/member/plugins/action/service.ts

```diff
@@ -1,15 +1,70 @@
-import { CannotModifyOtherMembers } from '../../../../utils/errors';
+import { PermissionLevel } from '@graasp/sdk';
+
+import { CannotModifyOtherMembers, UnauthorizedMember } from '../../../../utils/errors';
 import { Repositories } from '../../../../utils/repositories';
 import { ActionService } from '../../../action/services/action';
+import { validatePermissionMany } from '../../../authorization';
+import { Item, ItemExtraMap } from '../../../item/entities/Item';
 import { Actor } from '../../entities/member';
 
+export const getMonthBeforeNow = () => {
+  const date = new Date(); // Today's date
+  date.setMonth(date.getMonth() - 1); // Set the date to one month ago
+  return date;
+};
+
 export class ActionMemberService {
   actionService: ActionService;
 
   constructor(actionService: ActionService) {
     this.actionService = actionService;
   }
 
+  async getFilteredActions(
+    actor: Actor,
+    repositories: Repositories,
+    filters: { startDate?: string; endDate?: string },
+  ) {
+    const { actionRepository } = repositories;
+
+    const { startDate, endDate } = filters;
+    const start = startDate ? new Date(startDate) : getMonthBeforeNow();
+    const end = endDate ? new Date(endDate) : new Date();
+
+    if (!actor) {
+      throw new UnauthorizedMember(actor);
+    }
+    const [actionsNeedsPermission, actionsWithoutPermssion] = await Promise.all([
+      actionRepository.getMemberActionsNeedPesrmission(actor.id, {
+        startDate: start,
+        endDate: end,
+      }),
+      actionRepository.getMemberActionsNoNeedForPesrmission(actor.id, {
+        startDate: start,
+        endDate: end,
+      }),
+    ]);
```

Comment:
I think it's faster to get all the data, and then get all the items from the actions. What do you think? It can be expensive to fetch from the db.

Reply:
I think the initial idea was that there were actions that we would retrieve in any case, and others that we would need to filter based on user permissions. 

So I proposed to have 2 calls to the db that would reduce the logic we would have to do in JS. This may not been the best solution however, I would be open to seeing some tests on bottlenecks and where to place the divide for optimal performance, I also think that the code should read as clearly as possible. Even if we encoure a slight perf penalty.

Reply:
using this way is more readable somehow, and has less of work with js, as retrieving all data would require classifying actions based on what requires permission and what does not, then validating permission for those that require permission.

Reply:
From the where conditions I'm quite sure we don't have an index on 4 props, so it will basically be 2 linear scan of the db. I think it will be less hard the db if we do only one call, but I might be wrong. If we do a partition https://www.geeksforgeeks.org/lodash-_-partition-method/ I think it's more or less the same code.